### ISSUE TITLE
Does not get git user.email value if defined within an includeif gitconfig file

### ISSUE THREAD

**Issue by doolio** [mitsuhiko/rye#435]
`rye` already initialises the project directory as a git repository - great. I don't know the order in which it does this but could it use a user's git name and email settings and include those in the pyproject.toml it creates?

**Comment by CharlesChen0823** [comment]
rye using git config user and email, or you can set author in config.toml file(which describes in document)

**Comment by doolio** [comment]
> rye using git config user and email

Right you are but not in all cases it seems. See below. I've changed the issue title accordingly.

![image](https://github.com/mitsuhiko/rye/assets/13859009/f853b54c-a286-4311-80d3-92b2fb43a22e)

**Comment by CharlesChen0823** [comment]
can you show the output `git config --get-regexp "^user.(name|email)$"`

**Comment by doolio** [comment]
That seems to work.

![image](https://github.com/mitsuhiko/rye/assets/13859009/236eef83-f458-4b51-98a4-69ec3a5ee53f)

**Comment by CharlesChen0823** [comment]
I can't reproduce, can you show the steps of reproduce?

**Comment by doolio** [comment]
Of course. So firstly, my `git` config is as follows such that any repository in the "github" directory will use the github associated `user.email`.

![image](https://github.com/mitsuhiko/rye/assets/13859009/e07cacf9-10e5-44d5-aa74-505de850295e)

Steps to reproduce:
1. Create a directory for a `rye` managed python project
2. `cd` into this directory
3. Execute `rye init` which initialises the project directory as a git repository - great
4. Check `user.email` setting in this repository which should be set according to my `git` config - it is
5. Check `email` setting in the `pyproject.toml` expecting it to match `user.email` setting - it does not

![image](https://github.com/mitsuhiko/rye/assets/13859009/76c5b485-5660-4114-acb7-e74f0e5cdbde)

**Comment by CharlesChen0823** [comment]
Thanks, I reproduce.  
In my test, currently `includeIf` only work at an exists git repository.  
When create new project, `rye init`  firstly generate `pyproject.toml`,  then try execute `git init`; so `git` command cannot get the  content of `includeIf` at the steps of generating `pyproject.toml`.

**Comment by doolio** [comment]
> When create new project, `rye init` firstly generate `pyproject.toml`, then try execute `git init`; so `git` command cannot get the content of `includeIf` at the steps of generating `pyproject.toml`.

OK, so it could only work if `rye init` produces the `pyproject.toml` last. Is that a possibility? I would think many people use different emails with `git` and so don't set at the global level.

**Comment by CharlesChen0823** [comment]
I have try fix this problem [includeif not work](https://github.com/CharlesChen0823/rye/commit/aef3c521184c4cf17150b4adff5abfcb2e7264c6), but not sure this is a better.

**Comment by doolio** [comment]
@mitsuhiko thanks I hadn't but I removed my binary install and installed the latest version via `cargo`. However, your suggested command does not work. It returns unrecognized subcommand error which tallies with `update` not being a current subcommand of the latest version of `rye` or at least not the version I installed via `cargo`. Did you perhaps mean:

````
rye self update --rev e7d070402b5fdb2a73e92d3e1f09e530841fec8b
````
I went ahead an tried that in any case and it finished with the following output which I take means it is now at this commit.

![image](https://github.com/mitsuhiko/rye/assets/13859009/c001df6b-022c-4ae0-8c8a-b49b4e32c765)

Testing this commit I can state it is doing as I would like in using the `user.email` setting from an `includeif` git config file within the pyproject.toml. I hope the re-ordering of the steps performed by `rye init` to allow this don't have any unforeseen consequences as I think this is a small but very helpful feature that users that use different emails with git would appreciate. I personally would be very happy to see this commit merged into `main`. Thanks for looking into this and providing a solution @CharlesChen0823.

![image](https://github.com/mitsuhiko/rye/assets/13859009/e46d6511-e2f4-4ef6-bf49-db4fa03a0a1c)
